Fix usage of PTH_CONFIG to work with pthsem

The pthsem re-implement of pth provides a compatibility layer for pth,
but its pth-config script behaves slightly differently than the
original one when reporting the version number. This patch to gnupg2's
configure script adjusts the version checking to support this
difference, since Buildroot uses pth-config from pthsem.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
[yann.morin.1998@free.fr: adapt to patch m4 macro for autoreconfiguring]
Signed-off-by: "Yann E. MORIN" <yann.morin.1998@free.fr>

diff -durN gnupg2-2.0.29.orig/m4/gnupg-pth.m4 gnupg2-2.0.29/m4/gnupg-pth.m4
--- gnupg2-2.0.29.orig/m4/gnupg-pth.m4	2015-09-08 14:39:24.000000000 +0200
+++ gnupg2-2.0.29/m4/gnupg-pth.m4	2015-12-16 18:30:54.336513493 +0100
@@ -17,7 +17,7 @@
 # Taken and modified from the m4 macros which come with Pth.
 AC_DEFUN([GNUPG_PTH_VERSION_CHECK],
   [
-    _pth_version=`$PTH_CONFIG --version | awk 'NR==1 {print [$]3}'`
+    _pth_version=`$PTH_CONFIG --version | awk 'NR==1 {print [$]2}'`
     _req_version="ifelse([$1],,1.2.0,$1)"
 
     AC_MSG_CHECKING(for PTH - version >= $_req_version)
